src/Web/Tailwind.hs

```diff
@@ -43,23 +44,34 @@ import UnliftIO (MonadUnliftIO, finally)
 import UnliftIO.Directory (removeFile)
 import UnliftIO.Process (callProcess)
 import UnliftIO.Temporary (withSystemTempFile)
+import qualified Data.Text as Text
+
+data OfficialPlugin
+  = PluginTypography
+  | PluginForms
+  | PluginLineClamp
+  | PluginAspectRatio
+
+instance Text.Show.Show OfficialPlugin where
+  show = \case
+    PluginTypography  -> "typography"
+    PluginForms       -> "forms"
+    PluginLineClamp   -> "line-clamp"
+    PluginAspectRatio -> "aspect-ratio"
+
+mkStrPlugin :: OfficialPlugin -> Text
+mkStrPlugin plugin = "require('@tailwindcss/" <> Text.pack (show plugin) <> "')"
 
 -- | Haskell version of tailwind.config.js
 --
 -- Only the subset we care to define, as some fields (eg: plugins) are defined
 -- with arbitrary JS code.
 data TailwindConfig = TailwindConfig
   { -- | List of source patterns that reference CSS classes
-    _tailwindConfigContent :: [FilePath]
+    _tailwindConfigContent :: [FilePath],
+    _tailwindConfigPlugins :: [OfficialPlugin]
   }
   deriving (Generic)
-  deriving
-    (ToJSON)
-    via CustomJSON
-          '[ FieldLabelModifier
-               '[StripPrefix "_tailwindConfig", CamelToSnake]
-           ]
-          TailwindConfig
```

Comment:
Why was this removed?

Reply:
We can do something like this:

```haskell
data Plugin
  = Typography
  | Forms
  | LineClamp
  | AspectRatio
  deriving (Generic)
  deriving (ToJSON, FromJSON)
    via CustomJSON '[ ConstructorTagModifier '[ CamelToKebab ] ] Plugin

data TailwindConfig = TailwindConfig
  { -- | List of source patterns that reference CSS classes
    _tailwindConfigContent :: [FilePath],
    _tailwindConfigPlugins :: [Plugin],
    _tailwindConfigTheme   :: Aeson.Object
  }
  deriving (Generic)
  deriving
    (ToJSON)
    via PrefixedSnake "_tailwindConfig" TailwindConfig

instance Text.Show.Show TailwindConfig where
  show (decodeUtf8 . encode -> config) =
    toString
      [text|
      module.exports = ${config}
      |]
```

But `tailwind.config.js` isn't really JSON, it contains JavaScript.
So the `ToJSON` instance isn't actually that helpful at all.

I don't have a good idea, though. The `[text| ...]` template works fine, though.
We can patch the file together.

----

What is your opinion on having the `theme` object as `Aeson.Object` in `TailwindConfig`?
I added a command-line argument to add some file "theme.json", which is then JSON-decoded.

Reply:
Oh, okay. I thought it is used in Emanote, but if not - we can get rid of the json instance.

> What is your opinion on having the theme object as Aeson.Object in TailwindConfig?
I added a command-line argument to add some file "theme.json", which is then JSON-decoded.

Useful, but let's do it in separate PR, while keeping this PR focused.

Reply:
> we can get rid of the json instance.

I was wrong, sorry. See https://github.com/srid/tailwind-haskell/pull/13#discussion_r1142842486